frontend/src/api/k8s/roleBindings.ts

```diff
@@ -46,26 +46,28 @@ export const generateRoleBindingServingRuntime = (
   return roleBindingObject;
 };
 
-export const generateRoleBindingProjectSharing = (
+export const generateRoleBindingPermissions = (
   namespace: string,
-  rbSubjectType: ProjectSharingRBType,
+  rbSubjectType: RoleBindingPermissionsRBType,
   rbSubjectName: string,
-  rbRoleRefType: ProjectSharingRoleType,
+  rbRoleRefType: RoleBindingPermissionsRoleType | string, //string because with MR this can include MR name
+  rbRoleRefKind: string,
```

Comment:
This should be `rbRoleRefKind: 'Role' | 'ClusterRole'`

Reply:
i had that originally, but then noticed that the `RoleBindingSubject.kind` type was `string`, so thought I should match that

Reply:
That's correct, and that's a great catch, rolebindings could only be `Role` or `ClusterRole`, so we should limit

Reply:
@lucferbux - should i change the `RoleBindngSubject.kind` type in `k8sTypes`?

Reply:
@gitdallas yup, that should be the way to go, we should be mindful of our types, and try to be as precise as possible.

Reply:
@lucferbux - actually, do we need `RoleBindingSubject` in both both `frontend/src/types.ts` and `frontend/src/k8sTypes.ts`?

Reply:
Oh no, not at all, good catch, we should delete all duplicates, it could happen when the codebase grows that large.

Keep the one in `k8sTypes`